STRUCTURING_YOUR_PROJECT.md

```diff
@@ -1,5 +1,161 @@
 # Structuring Your Project
 
+This page outlines good practices for Python based projects, but many of the principles discussed here can be applied to 
+other software projects as well.
+
+## Start from a common skeleton which separates key package components
+
+Packages should rely on a common file structure which separates the various elements of a package into separate 
+locations. Remember that a package is not just the code needed to carry out science, but also documentation and 
+deployment. We recommend the following skeletal file structure for some package `examplepackage`:
+ 
+```
+devtools/
+  README.md
+  travis-ci/
+    install_miniconda.sh
+  appveyor-ci/
+    run_with_env.cmd
+  conda-recipe/
+    meta.yaml
+    build.sh
+    bld.bat
+    README.md
+docs/
+examplepackage/
+  tests/
+    __init__.py
+    test_examplepackage.py
+  __init__.py
+  examplepackage.py
+LICENSE.md
+README.md
+setup.py
+.travis.yml
+.appveyor.yml
+.gitignore
+```
+
+This list may appear daunting at first, but each element is either a critical component of what will make the software 
```

Comment:
We could likely make it less daunting by setting up an `examplepackage` repo with all of this boilerplate already filled in. Or we could write a Python script that sets up this boilerplate for them, asking a few simple questions (or reading from a YAML file).

Reply:
But it's still important to explain why we need all of this! So this is still super useful.

Reply:
I'll make a new issue for it, we can address some question I have for it there better and adjust this page once we have the boilerplate in place.

Reply:
A compchem focused [cookiecutter](https://github.com/audreyr/cookiecutter) repo would be awesome!

Reply:
Great idea! I have added the idea to the discussion in #54 as we figure it out.